lib/browser/rpc-server.js

```diff
@@ -449,3 +450,13 @@ ipcMain.on('ELECTRON_BROWSER_WINDOW_CLOSE', function (event) {
   }
   event.returnValue = null
 })
+
+ipcMain.on('ELECTRON_BROWSER_SANDBOX_LOAD', function (event, preloadPath) {
```

Comment:
I would not pass the `preloadPath` as an argument. We know which webContents sent the IPC so it should be possible to get the preload script path from the webPreferences.

Reply:
Apart from this issue I like this change, batching multiple synchronous IPC calls into make to make loading faster is definitely welcome 👍

Reply:
We are actually planning to add some filtering for `remote.require`, `remote.getGlobal`, etc.

Reply:
In my case I had to keep the preloadPath as renderer argument as I need to display an error in case the preload script cannot be loaded. How do you handle that case? Currently we get an error in JS console with the path to the preload script.

Reply:
> We are actually planning to add some filtering for remote.require, remote.getGlobal, etc.

Cool, that would be a welcome feature :+1: 

> In my case I had to keep the preloadPath as renderer argument as I need to display an error in case the preload script cannot be loaded. How do you handle that case? Currently we get an error in JS console with the path to the preload script.

`ELECTRON_BROWSER_SANDBOX_LOAD` can return an error if it fails to load the script

Reply:
Returning an error would be great. We should not have a silent failure without any logs in the console.

Reply:
Also if you just let fs.readFileSync throw and not assign the returnValue, sendSync in the renderer process will get stuck waiting.

Reply:
I agree with @miniak this is basically a one-liner "read any file plz" API, I'd rather we (in this PR) used `event.sender.getLastWebPreferences()`

Reply:
@tarruda this is the last remaining issue

Reply:
> I agree with @miniak this is basically a one-liner "read any file plz" API, I'd rather we (in this PR) used event.sender.getLastWebPreferences()

We currently give access to `electron.remote.fs` which can not only read any file, but also invoke any FS function. There's no regression in terms of security here by allowing the renderer to specify the preloadPath.

These are my arguments for keeping the current behavior and addressing this issue in a separate PR:

- If we start using `event.sender.getLastWebPreferences()`, there's no  reason to pass the preload path as argument to the renderer command line, so it would become necessary to refactor more code in C++ both in browser/renderer. (Unless we are OK with leaving dead code to be handled in a separate PR)
- There's no guarantee that this will work as expected without introducing new regressions. Maybe it will, but I will have to investigate if `preload` will be present in every `event.sender.webPreferences` (in windows created by window.open, for example, which follow a different webContents creation path).
- This is a completely separate change, not related to the goal of this PR which is to fix  #12316. I'd rather push refactoring changes in different PRs.

If you guys still insist that this should be done in the current PR, I will do it, but not right now since it will require more testing/verification which I can't do this week.

Reply:
@MarshallOfSound? I guess I am ok if this gets addressed in a separate PR.